Package.swift

```diff
@@ -1,52 +1,33 @@
-// swift-tools-version:5.0
+// swift-tools-version:5.5
```

Comment:
I'm also fine with using 5.0 and removing the platforms requirements, using `#if swift` `@available` on the task-local property instead. What do you think?

Reply:
Either way, the CI would need to be updated to account for 5.5 compatibility.

Reply:
Yeah so for better or worse let's stay as low as we can, to make adoption in other libs painless -- most *actual use* will indeed use the 5.5 task local, but it does not have to require it I suppose... Let's make adoption easier by allowing 5.0 -- even if the story for libraries will be "tracing only works if on 5.5".

Reply:
+1 to keep lower

Reply:
Okay, let's do 5.2 then. This will also come in handy for `SpanAttributes` in the tracing repo as we're using `@dynamicMemberLookup` there which was introduced in 5.2 👍

Reply:
NIO requires 5.2 now. So I guess 5.2 is very sensible.

Reply:
Thanks for chiming in Fabian :) Yep that sounds like a good baseline nowadays!